data/update.py

```diff
@@ -185,69 +160,66 @@ def scan(options):
     full_command += ["--serial"]
 
   # In real mode, ignore cached data, and parallelize.
-  else:
+  elif not options.get("cache"):
     full_command += ["--force"]
 
   shell_out(full_command)
 
-# Use domain-scan to gather .gov hostnames from public sources.
-# Then run pshtt on each gathered hostname.
-def subdomains(options):
+# Use domain-scan to gather .gov domains from public sources.
+def gather_subdomains(options):
+  print("[gather] Gathering subdomains.")
 
-  # Use domain-scan to gather .gov domains from public sources.
-  def gather_subdomains(gatherer, command):
-    print("[%s][gather] Gathering subdomains." % gatherer)
+  full_command = [GATHER_COMMAND]
 
-    gatherer_output = os.path.join(GATHER_TARGET, gatherer)
+  full_command += [",".join(GATHERER_NAMES)]
+  full_command += GATHERER_OPTIONS
 
-    full_command = [GATHER_COMMAND]
-    full_command += command
+  # Common to all gatherers.
+  # --parents gets auto-included as its own gatherer source.
+  full_command += [
+    "--output=%s" % SUBDOMAIN_DATA_GATHERED,
+    "--suffix=%s" % GATHER_SUFFIX,
+    "--ignore-www",
+    "--sort",
+    "--debug"
+  ]
 
-    # Common to all gatherers.
-    full_command += [
-      "--suffix=%s" % GATHER_SUFFIX,
-      "--output=%s" % gatherer_output,
-      "--parents=%s" % GATHER_PARENTS,
-      "--sort",
-      "--force",
-      "--debug"
-    ]
+  # In debug mode, allow cached data.
+  if (not options.get("debug")) and (not options.get("cache")):
+    full_command += ["--force"]
 
-    shell_out(full_command)
+  shell_out(full_command)
 
 
-  # Run pshtt on each gathered set of subdomains.
-  def scan_subdomains(gatherer):
-    print("[%s][scan] Scanning subdomains." % gatherer)
+# Run pshtt on each gathered set of subdomains.
+def scan_subdomains(options):
+  print("[scan] Scanning subdomains.")
 
-    subdomains = os.path.join(GATHER_TARGET, gatherer, "results", ("%s.csv" % gatherer))
-    scanner_output = os.path.join(SUBDOMAIN_SCAN_TARGET, gatherer)
+  subdomains = os.path.join(SUBDOMAIN_DATA_GATHERED, "results", "gathered.csv")
 
-    full_command = [
-      SCAN_COMMAND,
-      subdomains,
-      "--scan=%s" % SUBDOMAIN_SCANNERS,
-      "--output=%s" % scanner_output,
-      "--debug",
-      "--sort"
-    ]
+  full_command = [
+    SCAN_COMMAND,
+    subdomains,
+    "--scan=%s" % SUBDOMAIN_SCANNERS,
+    "--output=%s" % SUBDOMAIN_DATA_SCANNED,
+    "--sslyze-no-certs", # ugh, temporary
+    "--debug",
+    "--sort"
+  ]
 
-    # In debug mode, use cached data, and allow easy Ctrl-C.
-    if options.get("debug"):
-      full_command += ["--serial"]
+  # In debug mode, use cached data, and allow easy Ctrl-C.
+  if options.get("debug"):
+    full_command += ["--serial"]
 
-    # In real mode, ignore cached data, and parallelize.
-    else:
-      full_command += ["--force"]
+  # In real mode, ignore cached data, and parallelize.
+  elif not options.get("cache"):
+    full_command += ["--force"]
```

Comment:
Noting that in a future release of domain-scan, `--force` will become the default, and `--cache` will be needed to opt-in to making use of caching (which is only when `--debug` or `--cache` is passed to this script).